### PR TITLE
feat: enhance UI features and styling

### PR DESCRIPTION
**Pull request by artkolpakov**
This pull request implements several enhancements and adjustments to the user interface:

1. **Adjust Bar Chart Y-axis Styling**: improved the visual presentation of the bar chart by adjusting the Y-axis styling for better readability, aligning metrics with the chart's lines. [Ticket Nº107](https://github.com/Golf-Drill-Challenge-App/Golf-App/issues/107)

| example 1 | example 2 | example 3 |
|---------|---------|---------|
| Screenshot 2024-04-09 at 8 02 28 PM | Screenshot 2024-04-09 at 8 02 53 PM | Screenshot 2024-04-09 at 8 03 16 PM|

2. **Remove Email Update from User Profile Settings Modal:** removed the option for an email update from the user profile settings modal, as suggested by the coach. [Tiket Nº132](https://github.com/Golf-Drill-Challenge-App/Golf-App/issues/132).

Screenshot 2024-04-09 at 8 06 22 PM

3. **Display User's Email on Profile Card:** added functionality to display the user's email on the profile card. [Tiket Nº115](https://github.com/Golf-Drill-Challenge-App/Golf-App/issues/115)

Screenshot 2024-04-09 at 8 09 13 PM 

4. **Adjust Snapping Positions for Profile Settings Modal:** refactored the handling of snap points in the BottomSheetModal component to prevent full and half expansions. Additionally, ensured stability, as the snap points are now dynamically adjusted based on the visibility of the name/password input keyboard events (which wasn't working before). [Tiket Nº121](https://github.com/Golf-Drill-Challenge-App/Golf-App/issues/121)

| name update snap points | name + password update snap points | keyboard input |
|---------|---------|---------|
| IMG_6699 | IMG_6700 | IMG_6697 |

These changes collectively improve the usability and visual appeal of the application, enhancing the overall user experience.

### CODE REVIEW

File: package-lock.json
Comment (FrankreedX) [inline package-lock.json:1]: no lmao

File: components/barChart.js [hunk 1]
```diff
@@ -169,11 +169,11 @@ export default function BarChartScreen({ drillData, drillInfo }) {
     },
     yAxis: {
       position: "absolute",
-      top: 0,
+      top: -9.2,
```
Comment (FrankreedX) [inline components/barChart.js:172]: These changes clashes with my changes so prob remove them

**Review by FrankreedX — CHANGES_REQUESTED** [review]
overall pretty good!

**Review by FrankreedX — APPROVED** [review]

File: package-lock.json
Comment (artkolpakov) [reply package-lock.json:1]: Yeah, your yMin, and yMax calculations are much cleaner 😂, thanks!!

File: app/content/profile/index.js [hunk 2]
```diff
@@ -103,9 +102,30 @@ function Index(props) {
   }, [userData, userEmail]);
 
   useEffect(() => {
-    setSnapPoints(
-      passwordInputVisible ? expandedSnapPoints : initialSnapPoints,
+    setSnapPoints([passwordInputVisible ? "70%" : "57%"]);
+  }, [passwordInputVisible]);
+
+  useEffect(() => {
+    const keyboardDidShowListener = Keyboard.addListener(
```
Comment (solderq35) [inline app/content/profile/index.js:109]: I'll have to look at this more closely later but I think you can simplify a lot of this by using this library (we use it for keyboard inputs elsewhere, e.g. signin)

https://www.npmjs.com/package/react-native-keyboard-aware-scroll-view

**Review by solderq35 — CHANGES_REQUESTED** [review]
image

The signout button is still a little low imo (android, pixel 7 emulator)

image
- confirm password is unreachable on android and can't scroll lower (see other comment about react native keyboard aware scrollview library
  - Edit: that library seemed incompatible with bottomsheet

File: package-lock.json
Comment (solderq35) [reply package-lock.json:1]: Yes, please install yarn and regenerate yarn.lock (if you added or removed libraries)

**Review by solderq35 — COMMENTED** [review]
image

remember to rebase

**Comment by solderq35** [comment]
image

Might work for android "confirm password" issues, otherwise use https://reactnative.dev/docs/platform to add platform specific styling (keyboardawareview library doesn't seem compatible with bottomsheet)

**Comment by solderq35** [comment]
image

Increased vertical padding on Android so signout button is easier to see

**Review by solderq35 — APPROVED** [review]

**Pull request merged**